state-chain/pallets/cf-environment/src/tests.rs

```diff
@@ -226,3 +153,222 @@ fn update_safe_mode() {
 		));
 	});
 }
+
+#[test]
+fn can_discard_stale_utxos() {
+	let epoch_1 = [0xFE; 32];
+	let epoch_2 = [0xAA; 32];
+	let epoch_3 = [0xBB; 32];
+	let epoch_4 = [0xDD; 32];
+	new_test_ext().execute_with(|| {
+		// Do not discard utxo if `previous` key is not set.
+		CurrentBitcoinKey::set(Some(agg_key(epoch_2, None)));
+		ConsolidationParameters::<Test>::set(utxo_selection::ConsolidationParameters {
+			consolidation_threshold: 5,
+			consolidation_size: 2,
+		});
+
+		BitcoinAvailableUtxos::<Test>::set(vec![
+			utxo(1_000_000, CHANGE_ADDRESS_SALT, Some(epoch_1)),
+			utxo(2_000_000, CHANGE_ADDRESS_SALT, Some(epoch_1)),
+			utxo(3_000_000, CHANGE_ADDRESS_SALT, Some(epoch_1)),
+			utxo(31_000_000, CHANGE_ADDRESS_SALT, Some(epoch_3)),
+			utxo(32_000_000, CHANGE_ADDRESS_SALT, Some(epoch_3)),
+			utxo(33_000_000, CHANGE_ADDRESS_SALT, Some(epoch_3)),
```

Comment:
I feel like the tests can be more succinct/to the point. 

For example here, the important details are the epoch numbers, not the amounts or the change addresses. 

We could make this more obvious by not requiring this information to repeat, or by repeating the same amounts (then it's clear that the amounts don't matter).

Reply:
Simplified the unit tests